rust/connlib/tunnel/src/tests/transition.rs

```diff
@@ -83,6 +83,9 @@ pub(crate) enum Transition {
     ///
     /// In this case, we won't receive a `relays_presence` but instead we will receive relays with the same ID yet different credentials.
     RebootRelaysWhilePartitioned(BTreeMap<RelayId, Host<u64>>),
+
+    /// De-authorize access to a resource whilst the Gateway is network-partitioned from the portal.
```

Comment:
Out of curiosity, what does this test? That we respect the expiration timestamp of an authorization even while partitioned?

Reply:
Did you read the PR description? I tried to explain there what do test and what we don't test:

> We (partly) test this by introducing a new transition in our proptests that de-authorizes a certain resource whilst the Gateway is simulated to be partitioned. It is difficult to test that we cannot make a connection once that has happened because we would have to simulate a malicious client that knows about resources / connections or ignores the "remove resource" message.
> 
> Testing this is deferred to a dedicated task. We do test that we hit the code path of revoking the resource authorization and because the other resources keep working, we also test that we are at least not revoking the wrong ones.